File: python/graph/Vertex.py
class Vertex:
    def __init__(self, key):
        self.id = key
        self.connectedTo = {}

    def addNeighbour(self, v):
        self.connectedTo[v.id] = v
        return self.connectedTo
        
    def __str__(self):
        neighbours = ""
        for v in self.connectedTo.keys():
            neighbours += v + " "
        return str(self.id) + " -- " + neighbours

File: java/graph/Vertex.java
package graph;

import java.util.List;

public interface Vertex<T> {
    /**
     * Returns the actual content of the vertex
     *
     * @return the content of the vertex
     */
	public T getContent();

    /**
     * Returns the neighbours of the current vertex
     *
     * @return vertex neighbours
     */
	public List<Vertex<T>> getNeighbours();

    /**
     * Adds <code>v</code> as a new neighbour to the current vertex
     *
     * @param v new neighbour to be added
     */
    public void addNeighbour(Vertex<T> v);
}